### PR TITLE
feat(FX-3670): Enable the "save alert" button whenever the "Create Alert" screen has pending updates

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3670]

### Description
At the moment, the "save alert" button is blocked on the "Create Alert" screen if the user already has a previously saved alert with the same criteria. We need to make changes and make the button enable whenever the screen has pending updates

### Demo 1
https://user-images.githubusercontent.com/3513494/147357267-cd2526f6-588a-40ce-ac49-dbe4bb0f48df.mp4

### Demo 2
⚠️ please note that in this example, a previously created alert was updated (the name was updated), and **not a new one was created**

https://user-images.githubusercontent.com/3513494/147935401-8b5dbcbe-cc78-4134-b3d6-65a371957cb4.mp4

### Demo 3
A new alert has been created, since we do not have **a previously saved alert with the same criteria**

https://user-images.githubusercontent.com/3513494/147936102-2f140f11-60aa-4d83-a8f0-f88e4bdd44e2.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Enable the "save alert" button whenever the "Create Alert" screen has pending updates - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3670]: https://artsyproduct.atlassian.net/browse/FX-3670?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Enable the "save alert" button whenever the "Create Alert" screen has pending updates - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a3d29e7e88565a98abe3d2942d8194d701b58ee2
</p>

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ interface FormProps {
   artistName: string
   isLoading?: boolean
   isPreviouslySaved?: boolean
+  isChangedFilters?: boolean
```
Comment (dblandin) [inline src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:19]: **suggestion**: This might read a bit better as `hasChangedFilters?`

**Review by dblandin — APPROVED** [review]
One small non-blocking naming suggestion. LGTM!

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:19]: Yes, a good suggestion. Thanks!

**Pull request merged**